inst/report/scripts/run_molevolvr_pipeline.R

```diff
@@ -0,0 +1,1296 @@
+    ipr_lin <- ipr_lin %>%
+      # deselect the AccNum.y from the lineage table (y) and set
+      # the AccNum.x (x) from the ipr/acc2info tables to simply 'AccNum'
+      mutate(AccNum = AccNum.x) %>%
+      select(-AccNum.x, -AccNum.y)
+  }
+  # run add_name f(x) on ipr+lineage dataframe
+  ipr_lin <- ipr_lin %>%
+    addName() %>%
+    mutate(Name = gsub("^_", "", Name))
+
+  # add domarch info to iprscan + lineage df, only keep what's in x
+  ipr_cln <- left_join(ipr_lin, lookup_tbl, by = "DB.ID")
+
+  # populate empty description/short name columns
+  for (i in 1:nrow(ipr_cln)) {
+    if ((is.na(ipr_cln$ShortName[i]) || ipr_cln$ShortName[i] == "") &&
+        (is.na(ipr_cln$SignDesc[i]) || ipr_cln$SignDesc[i] == "-")) {
+
+      ipr_cln$SignDesc[i] <- ipr_cln$IPRDesc[i]
+      if (length(ipr_cln$LookupTblDesc[i]) != 0) {
+        ipr_cln$ShortName[i] <- ipr_cln$LookupTblDesc[i]
+      }
+    }
+    if (is.na(ipr_cln$ShortName[i]) || ipr_cln$ShortName[i] == "") {
+      ipr_cln$ShortName[i] <- ipr_cln$SignDesc[i]
+    }
+    if (is.na(ipr_cln$SignDesc[i]) || ipr_cln$SignDesc[i] == "-") {
+      ipr_cln$SignDesc[i] <- ipr_cln$ShortName[i]
+    }
+  }
+  # rename unclear/duplicated columns
+  names(ipr_cln)[names(ipr_cln) == "Description.x"] <- "ProteinName"
+  names(ipr_cln)[names(ipr_cln) == "Description.y"] <- "LookupTblDesc"
+  # deselect the AccNum.noV from the lineage table (y) and set
+  # the AccNum.noV.x (x) from the ipr/acc2info tables to simply 'AccNum.noV'
+  ipr_cln <- ipr_cln |> dplyr::select(-AccNum.noV.y) |>
+    dplyr::mutate(AccNum.noV = AccNum.noV.x) |>
+    dplyr::select(-AccNum.noV.x)
+  # create label column to use in ipr2viz
+  ipr_cln <- ipr_cln %>%
+    mutate(Label = strtrim(ShortName, 30)) %>%
+    mutate(Label = gsub(", .*", "", Label)) %>%
+    mutate(Label = gsub("C-terminal region of a signal",
+                        "C-term signal peptide", Label)) %>%
+    mutate(Label = gsub("N-terminal region of a signal",
+                        "N-term signal peptide", Label)) %>%
+    mutate(Label = gsub("Twin arginine translocation \\(T",
+                        "Tat signal profile", Label)) %>%
+    mutate(Label = gsub("GLUTATHIONE HYDROLASE PROENZYM",
+                        "GLUTATHIONE HYDROLASE PROENZYME", Label)) %>%
+    mutate(Label = gsub("N-terminal nucleophile aminohy",
+                        "Ntn hydrolases", Label)) %>%
+    mutate(Label = gsub("Region of a membrane-bound pro",
+                        "cytoplasmic reg of mem-bound prot",
+                        Label)) %>%
+    mutate(ShortName = gsub("Region of a membrane-bound protein predicted to be
+    outside the membrane, in the cytoplasm.",
+                            "cytoplasmic reg of mem-bound prot", ShortName))
+    outdir <- dirname(ipr)
+  # write results to file
+  write_tsv(ipr_cln, file.path(paste0(outdir, "/",paste0(prefix,".iprscan_cln.tsv"))))
+}
+
+ipr2da <- function(infile_ipr, prefix,
```

Comment:
Could avoid recalling function definitions. Just call the functions here as needed for the new functionalities.